webapp/package.json

```diff
@@ -94,7 +94,11 @@
     "react-transition-group": {
       "react": "17.0.2",
       "react-dom": "17.0.2"
-    }
+    },
+    "braces": "3.0.3",
+    "cheerio": "1.0.0-rc.12",
+    "@types/estree": "0.0.51",
+    "asn1.js": "5.4.1"
```

Comment:
these arent direct dependencies am i right? I am wondering if this should be done here like this or just have the lock file update them

Reply:
Yes, you are right. These packages are transitive dependencies. In this case I used the `overrides` to force specific, secure versions of these transitive deps. This was necessary because the parent packages (ie., `sass`, `enzyme`, `webpack`) have not updated their own dependencies to versions that solve the reported vulnerabilities. By definition of these overrides, we make sure that the vulnerable packages are updated to secure versions across the entire dependency tree.

Answering your question of why not let lock to handle it: by just updating the `package-lock.json` won't resolve this issue because the parent packages still specify the older and vulnerable versions of these transitive deps. Without `overrides`, running `npm ci` will  install the vulnerable versions as specified in the parent packages.

To keep a good track of what we are overriding we could start a Overrides.MD file where we define what, why and when to review again. Something like:

## Overrides Overview

| Package Name       | Overridden Version | Reason                                      | Date Added  | Review Date  |
|--------------------|--------------------|----------------------------------------------------------|-------------|--------------|
| `braces`           | `3.0.3`            | Fixes vulnerability                      | 2024-10-20  | 2025-07-20   |
| `cheerio`          | `1.0.0-rc.12`      | Addresses XSS vulnerability           | 2024-010-20  | 2025-07-20   |
| `@types/estree`    | `0.0.51`           | compatibility with updated TypeScript definitions | 2024-10-20  | 2025-07-20   |
| `asn1.js`          | `5.4.1`            | Mitigates Vulnerability             | 2024-10-20  | 2025-07-20   |

Thoughts @M-ZubairAhmed @hmhealey ?

Reply:
@hmhealey any comments for adding the overrides.md file to keep track of the used overrides? I think it will help to keep an eye on those and will serve as a reminder to at least review those once per quarter.

Reply:
Sorry, I missed this discussion earlier.

I don't know if we need that file since I feel like we're going to forget that it exists and it'll get out of date. There's a similar `.npmupgrade` file that's used by an upgrade tool we used for a bit (https://github.com/th0r/npm-upgrade), but I'm almost certain it's out of date too. I think this PR is sufficient for documenting why these overrides are the way they are

Reply:
ok, understood. But we need to somehow remember to review these overrides once in a while, specially run the npm ls {dep_name} and validate if we can upgrade our direct dependency and see if they move to a safe version of their direct dependency.